lib/ansible/plugins/connection/__init__.py

```diff
@@ -71,9 +71,8 @@ class ConnectionBase(AnsiblePlugin):
     def __init__(
         self,
         play_context: PlayContext,
-        new_stdin: io.TextIOWrapper | None = None,
-        shell: ShellBase | None = None,
         *args: t.Any,
+        shell: ShellBase | None = None,
```

Comment:
Do we care if a connection plugin was subclassed and they called `super().__init__(play_context, new_stdin, shell)`? I would have thought moving `shell` into `*args` then it could only be called with an explicit `shell=shell`.